src/cloud-api-adaptor/pkg/adaptor/proxy/service.go

```diff
@@ -140,50 +81,8 @@ func (s *proxyService) CreateContainer(ctx context.Context, req *pb.CreateContai
 		}
 	}
 
-	if pullImageInGuest {
-		logger.Printf("CreateContainer: Ignoring PullImage before CreateContainer (cid: %q)", req.ContainerId)
-	} else {
-		imageName, err := s.getImageName(req.OCI.Annotations)
-		if err != nil {
-			logger.Printf("CreateContainer: image name is not available in CreateContainerRequest: %v", err)
-		} else {
-			// Get the imageName from digest
-			if strings.HasPrefix(imageName, "sha256:") {
-				digest := imageName
-				logger.Printf("CreateContainer: get imageName from digest %q", digest)
-				imageName, err = s.getImageFromDigest(ctx, digest)
-				if err != nil {
-					return nil, err
-				}
-			}
-
-			logger.Printf("CreateContainer: calling PullImage for %q before CreateContainer (cid: %q)", imageName, req.ContainerId)
-
-			pullImageReq := &pb.PullImageRequest{
-				Image:       imageName,
-				ContainerId: req.ContainerId,
-			}
-
-			err = retry.Do(
-				func() error {
-					pullImageRes, pullImageErr := s.Redirector.PullImage(ctx, pullImageReq)
-					if pullImageErr != nil {
-						logger.Printf("CreateContainer: failed to call PullImage, probably because the image has already been pulled. ignored: %v", pullImageErr)
-						return pullImageErr
-					}
-					logger.Printf("CreateContainer: successfully pulled image %q", pullImageRes.ImageRef)
-					return nil
-				},
-			)
-
-			if err != nil {
-				logger.Printf("PullImage fails: %v", err)
-				return nil, err
-			}
-			// kata-agent uses this annotation to fix the image bundle path
-			// https://github.com/kata-containers/kata-containers/blob/8ad86e2ec9d26d2ef07f3bf794352a3fda7597e5/src/agent/src/rpc.rs#L694-L696
-			req.OCI.Annotations[cri.ImageName] = imageName
-		}
+	if !pullImageInGuest {
+		logger.Printf("Pulling image separately not support on main. It is required to use the nydus-snapshotter, which isn't configured properly here.")
```

Comment:
hmmm... what happens if CRI-O? `pullImageInGuest` will be `false` IIUC.

Reply:
So my understanding is the CRI-O still sets the storage driver to `image_guest_pull` as that is what the kata-agent is processing, so I think it should still set `pullImageInGuest` to be `true`. @littlejawa - can you confirm?

Reply:
Yes,  `image_guest_pull` is set by CRI-O, so we should be good here

Reply:
Thanks for the confirmation @littlejawa !